packages/runtime/container-runtime/src/pendingStateManager.ts

```diff
@@ -248,6 +273,26 @@ export class PendingStateManager implements IDisposable {
         }
     }
 
+    public checkpoint() {
+        const checkpointHead = this.pendingStates.peekBack();
+        return{
+            rollback:()=>{
+                try{
+                    while(this.pendingStates.peekBack() !== checkpointHead) {
+                        this.rollbackNextPendingState();
+                    }
+                } catch(err) {
```

Comment:
Trying to explore the idea we discussed of returning known failures rather than throwing.  So as-is, this `err` is thrown by  `this.rollbackNextPendingState()` and the only interesting thing in there is `this.stateHandler.rollback`.  What would it look like to have `rollback` on that interface return a strongly-typed failure result?